typegate/src/engine.ts

```diff
@@ -285,14 +339,45 @@ export class Engine {
           }, ${JSON.stringify(res)}`,
         );
       }
-      const field = path[path.length - 1] as any;
+
+      const field = path[path.length - 1];
       if (node !== "") {
         lens.forEach((l: any, i: number) => {
-          l[field] = res[i];
+          // Objects are replaced by empty objects `{}`.
+          // It will be populated by child compute stages using values in `cache`.
+          l[field] = withEmptyObjects(res[i]);
```

Comment:
if it is replaced anyway later, why putting empty object?

Reply:
Objects always start empty. The fields are set by child compute stages (corresponding to the field selections in the query).

Reply:
Yes, but letting `l[field] = res[I]` would avoid creating new object and still do the same behaviour? In this part of the engine, we should avoid creating or manipulating object/values so we have the less overhead possible.

Reply:
The object might have fields that are not  selected by the query. Replacing with empty object will ensure that we only have the selected fields in the final result.